### ISSUE TITLE
Generic JWT auth method

### ISSUE THREAD

**Issue by mbrancato** [hvac/hvac#572]
For Vault, the auth methods for GCP, Azure, Kubernetes, and JWT (maybe more?) are all really using JWT login, would it make sense to:

1. Add generic JWT auth method support to HVAC?
1. Make the login method for GCP, Azure, and Kubernetes just be wrappers to the JWT login?

**Comment by viralpoetry** [comment]
@mbrancato hi, just to note it here also for other people, I was able to misuse `Kubernetes` auth method for this purpose. Not sure if all config parameters will work but I was able to setup basis JWT auth method and use it with `client.auth_kubernetes`
```
# some configuration
vault write auth/jwt/config \
    jwks_url="https://idp.acme.com/auth/realms/somerealm/protocol/openid-connect/certs" \
    default_role="demo"

client.auth_kubernetes(
  "demo", jwt_token, use_token=True, mount_point=...
 )
```

**Comment by briantist** [comment]
Yeah it's good to know that the existing methods can be misued for that purpose.

I think ideally though there should be a dedicated interface for JWT.

If the interfaces for kubernetes/GCP/Azure change over time, clients using JWT shouldn't be affected by that. 

To the OP's point, if a JWT method is added, kubernetes/GCP/Azure can be wrappers around it for as long as they all do the same thing.

Having a stable interface is important for downstream projects that have their own users too like the linked Ansible issue.

It seems high value and low lift imo.

**Comment by mbrancato** [comment]
Thanks @briantist this really captures my thoughts around it. I think I’m currently misusing the GCP login to achieve this.

**Comment by jeffwecan** [comment]
Got a start down this road with https://github.com/hvac/hvac/pull/613. Suppose we just need to genericize that logic to cover the additional JWT login use cases (outside of the JWT auth method in particular covered in the linked PR).
